include/gc_tiny_fl.h

```diff
@@ -50,7 +50,7 @@
         || defined(__s390x__) \
         || (defined(__x86_64__) && !defined(__ILP32__)) \
         || defined(__alpha__) || defined(__powerpc64__) \
-        || defined(__arch64__)
+        || defined(__arch64__) || defined(__riscv)
```

Comment:
I wonder if `defined(__riscv)` is correct in isolation here? After all, I *think* we're really adding "RISC-V in its CHERI incarnation"? [It would also be good to think about which bits will/should work under Morello as we'll surely want to add support for that in soon.]

Reply:
This is not necessarily a CHERI problem. It is more a 64-bit problem. When I first started, there was not FREEBSD and/or RISC-V port. This has now partially come.

So I got rid  of my changes for the mainline riscv+freebsd config changes. However, this bit was not done. Hence I have changed it. 

Having said that, it should be behind a risc-*64* switch. This is done in new commit [408f667b](https://github.com/capablevms/bdwgc/pull/1/commits/408f667bb59d4d389e44ff70d5631bf922413ed9) .

Reply:
>  When I first started, there was not FREEBSD and/or RISC-V port

Boehm has run on FreeBSD for at least a couple of decades, so I'm not sure the first part of that sentence is correct :) I can well believe it hasn't been ported to riscv though.

Reply:
Yes, that is what I meant. The FreeBSD port for RISC-V has only been merged recently. Boehm for FreeBSD for other architectures are available all over the code.